lib/matplotlib/__init__.py

```diff
@@ -400,12 +400,15 @@ def impl(args, regex, min_ver=None, ignore_exit_code=False):
         try:
             output = subprocess.check_output(
                 args, stderr=subprocess.STDOUT,
-                text=True, errors="replace")
+                text=True, errors="replace", timeout=30)
         except subprocess.CalledProcessError as _cpe:
             if ignore_exit_code:
                 output = _cpe.output
             else:
                 raise ExecutableNotFoundError(str(_cpe)) from _cpe
+        except subprocess.TimeoutExpired as _te:
+            msg = f"Timed out running {' '.join(args)}"
```

Comment:
I might actually prefer `args` to be output as-is, just in case someone is passing something with spaces as an argument, as that keeps that visible.

Reply:
fwiw there's cbook._pformat_subprocess for that purpose as well.